sys/kern/kern_umtx.c

```diff
@@ -4843,7 +4843,7 @@ umtx_thread_exit(struct thread *td)
 }
 
 static int
-umtx_read_uptr(struct thread *td, uintptr_t ptr, uintcap_t *res, bool compat32)
+umtx_read_uptr(struct thread *td, uintcap_t ptr, uintcap_t *res, bool compat32)
```

Comment:
Could we change the type of ptr (and the other ones, e.g. td_rb_list, etc.) to be void* __capability so that we don't get implicit uint64_t/uint32_t -> uintcap_t conversions?

Reply:
It was changed in 3aa932d4d24c11d3204f10899f0ab069039c71bb so I was just changing it back. It looks like we could change the type, although it could get messy as I don't think we can change the robust_list_offset type as it's part of a syscall interface.